src/Plugin/Create/Payload.php

```diff
@@ -94,34 +104,96 @@ public static function fromRequest(Request $request): static {
 	 * @throws GenericError
 	 */
 	public static function hasMatch(Request $request): bool {
+		/**
+		 * @var array{
+		 *       CREATE?: array{
+		 *           expr_type: string,
+		 *           not-exists: bool,
+		 *           base_expr: string,
+		 *           sub_tree?: array<array{
+		 *               expr_type: string,
+		 *               base_expr: string
+		 *           }>
+		 *       },
+		 *       TABLE: array{
+		 *           base_expr: string,
+		 *           name?: string,
+		 *           no_quotes: array{
+		 *               delim: bool,
+		 *               parts: array<string>
+		 *           },
+		 *           create-def?: bool,
+		 *           options?: array<array{
+		 *               expr_type: string,
+		 *               base_expr: string,
+		 *               delim: string,
+		 *               sub_tree?: array<array{
+		 *                   expr_type: string,
+		 *                   base_expr: string
+		 *               }>
+		 *           }>
+		 *       },
+		 *       LIKE: array{
+		 *           expr_type: string,
+		 *           table?: string,
+		 *           base_expr: string,
+		 *           no_quotes: array{
+		 *               delim: bool,
+		 *               parts: array<string>
+		 *           }
+		 *       }
+		 *   }|false $payload
+		 */
 		$payload = Payload::$sqlQueryParser::parse(
 			$request->payload,
 			fn($request) => (
-				stripos(
-					$request->error,
-					"P03: syntax error, unexpected tablename, expecting \$end near 'with data'"
-				) !== false
-				&& stripos($request->payload, 'create') !== false
+				stripos($request->payload, 'create') !== false
```

Comment:
I'm not sure about the context, but to protect against false positives, it's a good idea to check that the position (index) of the word "create" is === 0. This ensures that we're only checking for the presence of "create" in the entire payload, rather than just checking if the word "create" appears anywhere. This can help avoid false validation when some other value or text also contains the word "create".